src/mem/alloc.h

```diff
@@ -1264,31 +1265,42 @@ namespace snmalloc
       MEASURE_TIME(remote_dealloc, 4, 16);
       SNMALLOC_ASSERT(target->id() != id());
 
-      handle_message_queue();
-
       void* offseted = apply_cache_friendly_offset(p, sizeclass);
 
       // Check whether this will overflow the cache first.  If we are a fake
       // allocator, then our cache will always be full and so we will never hit
       // this path.
       size_t sz = sizeclass_to_size(sizeclass);
-      if ((remote.size + sz) < REMOTE_CACHE)
+      if (remote.capacity > 0)
       {
         stats().remote_free(sizeclass);
         remote.dealloc_sized(target->id(), offseted, sz);
         return;
       }
+
+      remote_dealloc_slow(target, p, sizeclass);
+    }
+
+    SNMALLOC_SLOW_PATH
+    void remote_dealloc_slow(
+      RemoteAllocator* target, void* offseted, sizeclass_t sizeclass)
```

Comment:
The call to this function, a few lines up, passes p, ie. the non-offseted address,  as the second argument. 

I would duplicate the call to `apply_cache_friendly_offset`, once on the fast path inside the `if (remote.capacity > 0)` branch and once here, so you don’t have to also do a `remove_cache_friendly_offset`

Reply:
Agreed. Fixed.  Thanks for the feedback.  I had thought offset would have lead to better code, but actually it didn't.